src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
It is. Every context is bound to one udev seat (physical seat), it even exposes seats as an object and it allows you to move devices between logical seats, which you can create for yourself. But a seat itself is nothing else then a group of devices sharing that seat and a matching libinput context.
(The only methods of a seat are `get_devices` and `get_context`.)


The are no seat capabilities, instead devices or device groups have certain capabilities and all those properties and you can only derive properties of a seat by receiving and merging the properties of all it's attached devices.